streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceiverSupervisorImpl.scala

```diff
@@ -89,6 +90,9 @@ private[streaming] class ReceiverSupervisorImpl(
         case StopReceiver =>
           logInfo("Received stop signal")
           stop("Stopped by driver", None)
+        case DeleteOldBatch(threshTime) =>
```

Comment:
Since each such attempt can delete multiple batches (any batches older than the threshold time), its better to name this `DeleteOldBatches`